### PR TITLE
Feat/cos item

### PR DESCRIPTION
**Pull request by pranavrd**

### CODE REVIEW

File: src/main/java/iudx/catalogue/server/authenticator/JwtAuthenticationServiceImpl.java [hunk 1]
```diff
@@ -230,6 +229,18 @@ public AuthenticationService tokenInterospect(
...
+  private Future<Boolean> isValidAdmin(JwtData jwtData) {
+    // TODO: cop_admin or cos_admin???
+    if (jwtData.getRole().equalsIgnoreCase("cop_admin")) {
+      return Future.succeededFuture(true);
+    } else if(jwtData.getRole().equalsIgnoreCase("admin")) {
+      // TODO: write logic for rs admin check
+    } else {
+      return Future.failedFuture("admin token required for this operation");
+    }
+    return null;
```
Comment (kailash) [inline src/main/java/iudx/catalogue/server/authenticator/JwtAuthenticationServiceImpl.java:241]: Use Promise and Future, avoid returning null from methods

File: src/main/java/iudx/catalogue/server/authenticator/KcAuthenticationServiceImpl.java [hunk 2]
```diff
@@ -120,6 +117,15 @@ Future<Boolean> isValidEndpoint(String endpoint) {
     return promise.future();
   }
 
+  private Future<Boolean> isValidAdmin(JwtData jwtData) {
```
Comment (kailash) [inline src/main/java/iudx/catalogue/server/authenticator/KcAuthenticationServiceImpl.java:120]: The duplicate method. `isValidAdmin` method is also there in `JwtAuthServiceImpl`, try to use the same method and avoid code duplication.

**Review by kailash — CHANGES_REQUESTED** [review]
As a suggestion, you can also try to move delegate, provider, and cos_admin authorization logic[only approved delegates can work on their items, or providers can update only their items] inside their respective classes DelegateAuthStrategy, ProviderAuthStrategy etc..

**Comment by pranavrd** [comment]
retest this please

**Comment by pranavrd** [comment]
retest this please

**Review by kailash — APPROVED** [review]

**Pull request merged**